Add support for API tokens

royerboat opened the pull request:
Case PIG-2786: Added an 'api_token' option to the constructor so that a user
will have an explicit place to use their API token. Updated the constructor
unit test to test the new option.

Created a new unit test to test authentication using an API token.

Updated the POD documentation for authentication.

Additionally, updated the TFA unit test since it was failing.

ColMelvin commented:
I would ping @bk2204 about the copyright errors.  We may need to update cplint.

royerboat commented:
The only things that cplint fails on are the Copyright errors (which I'm not sure about) and the use of `done_testing` (which is used in other tests in this project). I'm not sure these things need to be fixed or if passing cplint is required for this repo.

More details on the copyright check:
```
[warn:Copyright] Existing file 't/04-tfa-sessions.t' had the following copyright errors:
No lines at all were found in file t/04-tfa-sessions.t matching any of the lines of the standard 4-line copyright header
[info:Copyright] File “t/05-api-query_tokens.t”: Attempted to check for open source copyright header, but no exact match found.
[error:Copyright] New file 't/05-api-query_tokens.t' had the following copyright errors:
No lines at all were found in file t/05-api-query_tokens.t matching any of the lines of the standard 4-line copyright header
[info:Copyright] Copyright headers must conform to the following standard four-line copyright header template (white space not significant):
# cpanel - <path relative to repo root>            Copyright 2017 cPanel, Inc.
#                                                           All rights Reserved.
# copyright@cpanel.net                                         http://cpanel.net
# This code is subject to the cPanel license. Unauthorized copying is prohibited
```

bk2204 commented:
Please use the standard BSD license template that's in the other files for the new files as well.  That theoretically should cause cplint to handle it correctly, and if it doesn't, we can ignore it.

As far as done_testing, let's avoid it in new files, but we can keep it in existing files.

royerboat commented:
I've added the copyrights and replaced the usage of done_testing with an actual plan in the new test file. cplint was still giving me errors about the copyright, though I did copy the open source copyright block directly from the cplint source code.
